fixtures/app_with_previews/Tuist/Package.swift

```diff
@@ -2,6 +2,20 @@
 
 import PackageDescription
 
+#if TUIST
+    import ProjectDescription
+
+    let packageSettings = PackageSettings(
+        productDestinations: [
+            "ResourcesFramework": [
+                .iPad,
+                .iPhone,
+                .macCatalyst,
+            ],
+        ]
```

Comment:
Curious why this was added as a part of this PR?

Reply:
This fixture depends on the local swift package. As it is recognized as a local package, it has all destinations and the [BuildAcceptanceTestAppWithPreviews](https://github.com/tuist/tuist/blob/b9003a64fd1aab836d51fd9031142769a6ba3914/Tests/TuistAutomationAcceptanceTests/BuildAcceptanceTests.swift#L51-L57) test failed due to  multiplatform.
So, I turned it to be the same as the original state.

Reply:
Does it mean that, with the current changes, local packages would _always_ include support for all targets? That feels a bit like a breaking change. Can we still narrow down to platforms defined exclusively in the `Project.swift` manifests in the which is done in the [ExternalProjectsPlatformNarrowerGraphMapper](https://github.com/tuist/tuist/blob/main/Sources/TuistDependencies/Mappers/ExternalProjectsPlatformNarrowerGraphMapper.swift)

Reply:
I agree that it will be a breaking change. So, I looking for breakthrough to resolve it but unfortunately, we can't use the [ExternalProjectsPlatformNarrowerGraphMapper](https://github.com/tuist/tuist/blob/main/Sources/TuistDependencies/Mappers/ExternalProjectsPlatformNarrowerGraphMapper.swift) as is. Because dependent local package is no longer considered to external project or the `isExternal` property is false at least.

To resolve it, I have a question. For a target that have local packages as dependencies, do you think the target should be included to the return value of [targetsWithExternalDependencies()](https://github.com/tuist/tuist/blob/6864dd791daa3b54996d92ed971e4c6b80314ad6/Sources/TuistCore/Graph/GraphTraverser.swift#L674-L676)?

I believe this question will help to reuse [ExternalProjectsPlatformNarrowerGraphMapper](https://github.com/tuist/tuist/blob/main/Sources/TuistDependencies/Mappers/ExternalProjectsPlatformNarrowerGraphMapper.swift).

Otherwise, what do you think about adding a new mapper?

Reply:
> as is because dependent local package is no longer considered to external project or the isExternal property is false at least.

What do you think of adding the following enum and replacing the `isExternal` property with it:
```swift
enum ProjectType {
  case remotePackage
  case localPackage
  case tuistProject
}
```

We can use this enum to preserve the information of whether a project is coming from a package or not. The [ExternalProjectsPlatformNarrowerGraphMapper](https://github.com/tuist/tuist/blob/main/Sources/TuistDependencies/Mappers/ExternalProjectsPlatformNarrowerGraphMapper.swift) would then narrow platforms for all projects that are `remotePackage` or `localPackage` instead of just `isExternal`

Reply:
Your suggestion worked very well. 🙏 I'm going to update the PR soon.